my-app/src/containers/PersonalFinanceApp.tsx

```diff
@@ -0,0 +1,159 @@
+
+  const toggleMenu = () => {
+    setOpen(!open);
+  };
+
+  const handleItemClick = (component: string) => {
+    setSelectedComponent(component);
+  };
+
+  const components = {
+    overview: Overview,
+    pots: Pots,
+    budgets: Budgets,
+    transactions: Transactions,
+    recurringBills: RecurringBills,
+  };
+
+  const SelectedComponent = selectedComponent
+    ? components[selectedComponent]
+    : null;
+
+  return (
+    <StyledWrapper>
+      <Drawer
+        sx={{
+          width: open ? 0 : 60,
+          flexShrink: 0,
+          "& .MuiDrawer-paper": {
+            width: open ? 250 : 100,
+            transition: "width 0.3s",
+            backgroundColor: "#201F24",
+            borderTopRightRadius: "15px",
+            borderBottomRightRadius: "15px",
+            color: "#fff",
+          },
+        }}
+        variant="persistent"
+        anchor="left"
+        open={true}
+      >
+        <StyledWrapperTypography>
+          {open ? (
+            <img
+              src={`${process.env.PUBLIC_URL}/images/logo-large.svg`}
+              alt="Logo"
+            />
+          ) : (
+            <img
+              src={`${process.env.PUBLIC_URL}/images/logo-small.svg`}
+              alt="Logo"
+            />
+          )}
+        </StyledWrapperTypography>
+
+        <StyledWrapperList>
+          {Object.keys(menuItems).map((key) => {
+            const { text, image, description } = menuItems[key];
+            return (
+              <div key={key}>
+                <button onClick={() => handleItemClick(key)}>
+                  <img
+                    src={`${process.env.PUBLIC_URL}/images/${image}.svg`}
+                    alt={description}
+                    style={{ width: 24, height: 24 }}
+                  />
+                  {open && <p>{text}</p>}
+                </button>
+              </div>
+            );
+          })}
+        </StyledWrapperList>
+
+        <IconButton
+          onClick={toggleMenu}
+          style={{
+            position: "absolute",
+            bottom: 20,
+            left: "50%",
+            transform: "translateX(-50%)",
+          }}
+        >
+          {open ? (
+            <ButtonMinimizeMenu>
+              <img
+                src={`${process.env.PUBLIC_URL}/images/icon-caret-left.svg`}
+                alt="Caret left"
+              />
+              Minimize Menu
+            </ButtonMinimizeMenu>
+          ) : (
+            <ButtonMinimizeMenu>
+              <img
+                src={`${process.env.PUBLIC_URL}/images/icon-caret-right.svg`}
+                alt="Caret right"
+              />
+            </ButtonMinimizeMenu>
+          )}
+        </IconButton>
+      </Drawer>
+
+      <div
+        style={{
+          marginLeft: open ? 240 : 80,
+          padding: "20px",
+          flexGrow: 1,
+        }}
+      >
+        {SelectedComponent && <SelectedComponent />}
```

Comment:
```suggestion
        <SelectedComponent />
```

is enough. If `SelectedComponent` is `null` or `undefined` then simply nothing renders.